op-service/txmgr/txmgr.go

```diff
@@ -84,6 +87,13 @@ type SimpleTxManager struct {
 	backend ETHBackend
 	l       log.Logger
 	metr    metrics.TxMetricer
+
+	nonce     *uint64
+	lock      sync.RWMutex
+	wg        sync.WaitGroup
+	resetC    chan struct{}
+	resetWG   sync.WaitGroup
+	resetting atomic.Bool
```

Comment:
There's a fair few fields being added here to control various aspects of concurrency.  It would probably be helpful to add some comments to explain what they're actually used for.

It certainly is a lot of concurrency management all tied to the way reset works. If the `ErrGroup` idea @trianglesphere mentioned below could simplify things that would definitely be a win. The current code works, it's just hard to maintain code that has multiple overlapping concurrency controls like this because the interactions can be hard to reason about.

Reply:
thanks, agree 👍  I removed all of the concurrency management from `txmgr` except for the nonce + lock